main/src/ca/uwaterloo/flix/language/ast/LevelEnv.scala

```diff
@@ -23,14 +23,15 @@ sealed trait LevelEnv {
   /**
     * Purifies regions in the type if they are out of scope.
     */
-  def purify(tpe: Type): Type = this match {
-    case LevelEnv.Unleveled => tpe
-    case LevelEnv.Leveled(scopes) =>
-      tpe.map {
-        case Type.Var(sym, _) if (sym.isRegion && !scopes.contains(sym)) => Type.Pure
-        case t => t
-      }
-  }
+  def purify(tpe: Type): Type = tpe
```

Comment:
Remove function? Or remove LevelEnv entirely?

Reply:
We can remove LevelEnv. Do we want that here or in an immediate followup?

Reply:
Here. Why not?